internal/fixture/state.go

```diff
@@ -168,20 +169,47 @@ func (s *State) Apply(obj typed.YAMLObject, version fieldpath.APIVersion, manage
 
 // CompareLive takes a YAML string and returns the comparison with the
 // current live object or an error.
-func (s *State) CompareLive(obj typed.YAMLObject, version fieldpath.APIVersion) (*typed.Comparison, error) {
+func (s *State) CompareLive(obj typed.YAMLObject, version fieldpath.APIVersion) (string, error) {
 	obj = FixTabsOrDie(obj)
 	if err := s.checkInit(version); err != nil {
-		return nil, err
+		return "", err
 	}
 	tv, err := s.Parser.Type(string(version)).FromYAML(obj)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 	live, err := s.Updater.Converter.Convert(s.Live, version)
 	if err != nil {
-		return nil, err
+		return "", err
+	}
+	tvu := convertMapAnyToMapString(tv.AsValue().Unstructured())
+	liveu := convertMapAnyToMapString(live.AsValue().Unstructured())
+	return cmp.Diff(tvu, liveu), nil
```

Comment:
I'm sus about how `cmp.Diff` equality check works on the values here. I'd be more comfortable adding options to the existing `Compare` to get the desired behavior. Would also save us a conversion to unstructured for old/new on every write.

Reply:
well, this is a test here and cmp.Diff is convenient to see what's different. I can use a [`Comparer`](https://pkg.go.dev/github.com/google/go-cmp/cmp#Comparer) that uses `Value.Equals` if you think it's safer?

Reply:
ah did not realize this was a test. as you were